File: v1/frontend/lib/data/models/plan.dart
// lib/data/models/plan.dart

import 'package:freezed_annotation/freezed_annotation.dart';

part 'plan.freezed.dart';
part 'plan.g.dart';

@freezed
class TrainingPlan with _$TrainingPlan {
  const factory TrainingPlan({
    required String id,
    required String title,
    String? goal,
    int? durationWeeks,
    Map<String, dynamic>? planContent,
    @Default(true) bool isActive,
    DateTime? createdAt,
  }) = _TrainingPlan;

  factory TrainingPlan.fromJson(Map<String, dynamic> json) => _$TrainingPlanFromJson(json);
}

File: v1/backend/app/models/plan.py
# app/models/plan.py

from sqlalchemy import Column, String, Integer, Boolean, DateTime, Date
from sqlalchemy.dialects.postgresql import UUID, JSONB
import uuid
from datetime import datetime

from app.db.base import Base


class TrainingPlan(Base):
    __tablename__ = "training_plans"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id = Column(UUID(as_uuid=True), nullable=False)  # FK to users(id)
    title = Column(String(200), nullable=False)
    goal = Column(String(50))  # 'weight_loss', 'muscle_gain', 'health'
    duration_weeks = Column(Integer)

    plan_content = Column(JSONB)

    start_date = Column(Date)
    end_date = Column(Date)
    is_active = Column(Boolean, default=True)

    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
